plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1212,13 +1221,33 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
         try {
             long commitVersion = handle.getReadVersion() + 1;
 
+            ImmutableMap.Builder<String, Map<String, Object>> columnMetadata = ImmutableMap.builder();
+            columnMetadata.putAll(getColumnsMetadata(handle.getMetadataEntry()));
+            ImmutableMap.Builder<String, Object> metadata = ImmutableMap.builder();
+            Map<String, String> configuration = new HashMap<>(handle.getMetadataEntry().getConfiguration());
+            AtomicInteger newFieldId = new AtomicInteger();
+            String physicalColumnName;
+            switch (columnMappingMode) {
+                case NONE -> {
+                    physicalColumnName = newColumnMetadata.getName();
+                }
+                case ID, NAME -> {
+                    int maxColumnId = getMaxColumnId(handle.getMetadataEntry());
+                    verify(maxColumnId != Integer.MAX_VALUE);
+                    newFieldId.set(maxColumnId + 1);
+                    physicalColumnName = generatePhysicalColumnName();
+                }
+                default -> throw new IllegalArgumentException("Unexpected column mapping mode: " + columnMappingMode);
+            }
+            columnMetadata.put(newColumnMetadata.getName(), metadata.buildOrThrow());
```

Comment:
`metadata` seems to be unused - it gets initialized with `ImmutableMap.builder()` and then I see only `buildOrThrow()` called on it.